lib/iris/fileformats/pp_rules.py

```diff
@@ -902,6 +905,22 @@ def _all_other_rules(f):
             (AuxCoord('son', long_name='season', units='no_unit'),
              None))
 
+    # Special case where year is zero and months match.
+    # Month coordinates (--> scalar coordinates)
+    if (f.lbtim.ib == 2 and f.lbtim.ic in [1, 2, 4] and
+            ((len(f.lbcode) != 5) or
+             (len(f.lbcode) == 5 and
+              f.lbcode.ix not in [20, 21, 22, 23] and
+              f.lbcode.iy not in [20, 21, 22, 23])) and
+            f.lbyr == 0 and f.lbyrd == 0 and
+            f.lbmon == f.lbmond):
+        aux_coords_and_dims.append(
+            (AuxCoord(f.lbmon, long_name='month_number', units='no_unit'),
```

Comment:
i thought the intent of the last commit was to input the string of the month, similar to the behaviour of
http://scitools.org.uk/iris/docs/latest/iris/iris/coord_categorisation.html?highlight=categorisation#iris.coord_categorisation.add_month
I think this is helpful, but it does preclude the ordering based on this coordinate

may I suggest
```
import calendar

        aux_coords_and_dims.append(
            (DimCoord(f.lbmon, long_name='month_number', units='no_unit'), None))
        aux_coords_and_dims.append(
            (AuxCoord(calendar.month_abbr[f.lbmon], long_name='month', units='no_unit'), None))